### PR TITLE
Gulp setup

### PR DESCRIPTION
**Pull request by brianlee389**

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -0,0 +1,76 @@
+Setting Up Gulp
```
Comment (thomaspark) [inline README:1]: Not sure why this file is here.

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+gulp.task('default', ['minify-snowball-css', 'minify-snowball-js', 'minify-templates', 'minify-admins'], function () {
+  gulp.watch("scripts/*.js", ['minify-snowball-js']);
+  gulp.watch("styles/*.css", ['minify-snowball-css']);
+  gulp.watch("modules/*/template.js", ['minify-templates']);
+  gulp.watch("modules/*/admin.js", ['minify-admins']);
+});
+
+gulp.task('minify-snowball-css', function () {
+  //var negateMinCSS = 'styles/!(*.min.css)'; for retrieving all non .min.css
```
Comment (thomaspark) [inline gulpfile.js]: Can we get rid of these `var negateMinCSS` lines now?

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  gulp.src([dir + '*.css', '!' + dir + '*.min.css'])
+    .pipe(plumber({
+      errorHandler: onError
+    }))
+    .pipe(minifyCss())
+    .pipe(rename(function (path) {
+      path.basename += '.min';
+    }))
+    .pipe(gulp.dest("./lib/fixed-sticky/"));
+});
```
Comment (thomaspark) [inline gulpfile.js:122]: newline

File: inc/snowball-editor.php [hunk 4]
```diff
@@ -24,6 +24,7 @@
   </div>
 
 <?php
+  echo '<script defer src="' . plugins_url('snowball/scripts/min/admins.min.js') . '"></script>';
```
Comment (thomaspark) [inline inc/snowball-editor.php]: Do we need this if we're already enqueuing it?

File: inc/snowball-enqueue.php [hunk 5]
```diff
@@ -28,21 +26,22 @@ function snowball_admin_add_scripts_and_stylesheets($hook) {
...
 
     wp_enqueue_script('media-upload');
     wp_enqueue_script('thickbox');
-    wp_enqueue_script('codemirror-js', plugins_url('snowball/lib/codemirror/lib/codemirror.js'));
-    wp_enqueue_script('codemirror-mode-javascript', plugins_url('snowball/lib/codemirror/mode/javascript.js'));
-    wp_enqueue_script('codemirror-mode-css', plugins_url('snowball/lib/codemirror/mode/css.js'));
-    wp_enqueue_script('codemirror-mode-xml', plugins_url('snowball/lib/codemirror/mode/xml.js'));
-    wp_enqueue_script('fixedsticky-js', plugins_url('snowball/lib/fixed-sticky/fixedsticky.js'), array('jquery'), '', true);
+    wp_enqueue_script('codemirror-js', plugins_url('snowball/lib/codemirror/lib/codemirror.min.js'));
+
```
Comment (thomaspark) [inline inc/snowball-enqueue.php]: Get rid of blank line.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (thomaspark) [inline package.json:1]: package.json is metadata for the project, not just the gulp file. Should reflect the main repo.

**Comment by thomaspark** [comment]
@brianlee389, this looks really good. Main issues are a random `README` file being added, `admins.min.js` being included in two places, and the `package.json` metadata being wrong. Also left a few smaller inline comments.

**Comment by brianlee389** [comment]
I wrote the README for how to use gulp

**Comment by thomaspark** [comment]
Oh I see, I thought it was added by Gulp somehow. If we're going to add a README, it should have general info about the project, with this just being one section instead of the whole file. You can leave it as is for now.

**Comment by brianlee389** [comment]
I rebased and pushed again

**Comment by thomaspark** [comment]
Nice work on this!

**Comment by brianlee389** [comment]
Thanks

**Pull request merged**